packages/create-sitecore-jss/src/templates/nextjs/src/components/SafeHydrate.tsx

```diff
@@ -0,0 +1,9 @@
+﻿import dynamic from 'next/dynamic';
+import React from 'react';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const NonSSRWrapper = (props: any) => <React.Fragment>{props.children}</React.Fragment>;
+
+export default dynamic(() => Promise.resolve(NonSSRWrapper), {
```

Comment:
Is `Promise.resolve` required  to be used by type?

Reply:
@illiakovalenko  Dynamically importing a component within a function returns a promise. If the module loads properly it would resolve the promise otherwise it will not.